cudet/updates.py

```diff
@@ -0,0 +1,94 @@
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+
+import subprocess
+import yaml
+import glob
+from copy import deepcopy
+from fuelclient.commands import base, environment
+from fuelclient.cli.actions.fact import DeploymentAction
+from fuelclient.cli.actions.settings import SettingsAction
+from fuelclient.cli.error import ServerDataException
+
+MU_UPGRADE_DICT_RESTRICTED = {
+    'mu_upgrade': {
+        'enabled': True,
+        'restart_rabbit': True,
+        'restart_mysql': True,
+    }
+}
+MU_UPGRADE_DICT_FULL = deepcopy(MU_UPGRADE_DICT_RESTRICTED)
+MU_UPGRADE_DICT_FULL['mu_upgrade'].update({
+    'metadata': {
+        'group': 'general',
+        'label': 'MU label',
+        'restrictions': [
+            {'action': 'hide'}
+        ],
+        'weight': 10
+    },
+    'type': 'hidden',
+    'value': True
+})
+
+
+class Updates(base.BaseCommand):
+    """TODO: provide meaningful description"""
+
+    def get_parser(self, prog_name):
+        parser = super(Updates, self).get_parser(prog_name)
+        parser.add_argument('env_id',
+                            type=str,
+                            help='ID of the environment')
+        parser.add_argument('install',
+                            type=bool, dest='do_install'
+                            help='Install MU')
+        return parser
+
+    def _update_settings_file(filename, full=True):
+        settings_yaml = None
+        with open(filename, 'r') as f:
+            settings_yaml = yaml.load(f.read())
+        settings_yaml['editable'].update(
+            MU_UPGRADE_DICT_FULL if full else MU_UPGRADE_DICT_RESTRICTED)
+        with open(filename, 'w') as f:
+            stream = yaml.dump(settings_yaml, default_flow_style=False)
+            f.write(stream.replace('{', ''))
+
+    def take_action(self, parsed_args):
+        if parsed_args.do_install:
+            try:
+                env_id = parsed_args.env_id
+                deployment_action = DeploymentAction()
+                settings_action = SettingsAction()
+                try:
+                    deployment_action.download(parsed_args)
+                except ServerDataException as e:
+                    if 'There is no deployment info for this environment' in e.message:
+                        settings_action.download(parsed_args)
+                        _update_settings_file('/root/settings_{0}'.format(env_id))
+                else:
+                    for filename in glob.glob("/root/deployment_{0}/*.yaml".format(env_id)):
+                        _update_settings_file(filename, full=False)
+                settings_action.upload(parsed_args)
+                env_redeploy = environment.EnvRedeploy()
```

Comment:
Somehow instantiation of the EnvRedeploy class is not working for me.
Could we be using something like
`task_id = self.client.redeploy_changes(parsed_args.id)`
and then print this task_id in output to help user find the right task in the list.
I checked this out and it works. Though, I don't understand why calling of `take_action` of EnvRedeploy does nothing...